pyop2/base.py

```diff
@@ -3575,7 +3564,8 @@ def compute(self):
             self.update_arg_data_state()
             for arg in reversed(self.args):
                 if arg._is_mat and arg.lgmaps is not None:
-                    arg.data.handle.setLGMap(*orig_lgmaps.pop())
+                    for m, lgmaps in zip(arg.data, orig_lgmaps.pop()):
+                        m.handle.setLGMap(*lgmaps)
```

Comment:
`orig_lgmaps.pop()` is just `olgmaps` as defined above right? It looks like whatever was previously contained in `orig_lgmaps` is thrown out either way (in fact, it basically only ever contains `olgmaps`).

I guess basically what I'm saying is the variable `orig_lgmaps` appears to do nothing. You gather `olgmaps` above, add it to the empty `orig_lgmaps` list, then extract `olgmaps` via `orig_lgmaps.pop()`.

Perhaps I'm just being dense and not seeing the reason for `orig_lgmaps`

Reply:
Now we have to restore the swapped out lgmaps for the original ones (so we don't always discard entries in these rows/columns). So we walk do that here.